lib/inspec/runner.rb

```diff
@@ -66,83 +80,114 @@ def load_attributes(options)
       options['attributes'] = attributes
     end
 
-    # Returns the profile context used the profile at this target.
+    #
+    # add_target allows the user to add a target whose tests will be
+    # run when the user calls the run method. This is the main entry
+    # point to profile loading.
+    #
+    # A target is a path or URL that points to a profile. Using this
+    # target we generate a Profile and a ProfileContext. The content
+    # (libraries, tests, and attributes) from the Profile are loaded
+    # into the ProfileContext.
+    #
+    # If the profile depends on other profiles, those profiles will be
+    # loaded on-demand when include_content or required_content are
+    # called using similar code in Inspec::DSL.
+    #
+    # Once the we've loaded all of the tests files in the profile, we
+    # query the profile for the full list of rules. Those rules are
+    # registered with the @test_collector which is ultimately
+    # responsible for actually running the tests.
+    #
+    # TODO: Deduplicate/clarify the loading code that exists in here,
+    # the ProfileContext, the Profile, and Inspec::DSL
+    #
+    # TODO: Libraries of dependent profiles should be loaded even when
```

Comment:
+1 right, that is essential for resource packs aka profiles without any test, but only libraries